### PR TITLE
Clippy-driven refactoring

### PR DESCRIPTION
**Pull request by tsionyx**
Added some changes from my [previous PR](https://github.com/lancastr/slog-gelf/pull/4) to allow to implement TCP transport later on top of it.

### CODE REVIEW

File: src/chunked.rs [hunk 1]
```diff
@@ -143,11 +143,11 @@ impl<'a> ChunkedMessageId {
     fn random() -> ChunkedMessageId {
         let mut bytes = [0; 8];
 
-        for b in 0..8 {
-            bytes[b] = rand::random();
+        for byte in &mut bytes {
+            *byte = rand::random();
```
Comment (glebpom) [inline src/chunked.rs]: `fill_bytes` from `rand` would be a better refactoring here

File: src/lib.rs [hunk 2]
```diff
@@ -10,23 +10,25 @@ extern crate serde;
...
+    destination: D,
 }
 
-impl Gelf {
+pub trait Destination {
+    fn log(&self, message: Vec<u8>) -> Result<(), io::Error>;
+}
+
+impl Gelf<UdpDestination> {
     pub fn new(source: &str, destination: &str) -> Result<Self, io::Error> {
```
Comment (glebpom) [inline src/lib.rs]: I think having two functions that do the same is not the best approach. I would suggest getting rid of `new` and keep `new_with_udp` and `new_with_tcp` in the future.

File: src/chunked.rs (hunk 1 above)
Comment (tsionyx) [reply src/chunked.rs]: done

**Review by glebpom — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 2 above)
Comment (tsionyx) [reply src/lib.rs]: I left it only for backward compatibility. May be we should mark the `new` method as `#[deprecated]` for now and then remove it in the future.

Comment (glebpom) [reply src/lib.rs]: Looks good to me, let's deprecate it

**Review by glebpom — APPROVED** [review]

**Pull request merged**